modules/basegl/src/algorithm/volumenormalization.cpp

```diff
@@ -0,0 +1,104 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2016-2021 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/basegl/algorithm/volumenormalization.h>
+
+#include <modules/opengl/volume/volumegl.h>
+#include <modules/opengl/shader/shaderutils.h>
+#include <modules/opengl/volume/volumeutils.h>
+
+namespace inviwo {
+
+VolumeNormalization::VolumeNormalization()
+    : shader_({{ShaderType::Vertex, utilgl::findShaderResource("volume_gpu.vert")},
+               {ShaderType::Geometry, utilgl::findShaderResource("volume_gpu.geom")},
+               {ShaderType::Fragment, utilgl::findShaderResource("volumenormalization.frag")}},
+              Shader::Build::No)
+    , fbo_()
+    , needsCompilation_(false) {
+    shader_.getFragmentShaderObject()->addShaderDefine("NORMALIZE_CHANNEL_0");
+    shader_.build();
+}
+
+void VolumeNormalization::setNormalizeChannel(const size_t channel, const bool normalize) {
```

Comment:
should there be a convenience function to set all 4 at once? e.g. with a `bvec4` or 4x bool

Reply:
I though about it. If you think it would be useful, I'll add it. Didn't want to overload the whole thing.